python/ray/serve/_private/deployment_state.py

```diff
@@ -1310,8 +1311,10 @@ def notify_running_replicas_changed(self) -> None:
         ):
             return
 
+        prefix = self.app_name + "_" if self.app_name else ""
```

Comment:
Here is the only place where there is information flow from deployment state --> routers, so we need to separate app name from deployment name here only.